build_tools/cmake/iree_static_linker_test.cmake

```diff
@@ -142,88 +220,177 @@ function(iree_static_linker_test)
-    list(POP_BACK _INPUT_ENTRY_LIST _INPUT_FORMAT)
+  iree_cc_binary(
+    NAME
+      ${_RULE_NAME}_run
+    SRCS
+      "${CMAKE_CURRENT_BINARY_DIR}/${_RULE_NAME}.c"
+    DEPS
+      ::${_RULE_NAME}_lib
+      iree::runtime
+      iree::hal::drivers::local_sync::sync_driver
+      iree::hal::local::loaders::static_library_loader
+  )
 
-    # Process shape to compute the size (number of elements) and dimention.
-    # The results are stored into lists to support multiple inputs
-    if(NOT _INPUT_ENTRY_LIST)  # single entry input
-      set(_INPUT_ENTRY_LIST "1")
-    endif()
+  if(_RULE_EMITC)
+    target_link_libraries(${_NAME}_run
+        PRIVATE
+          iree_vm_shims_emitc
+          ${_NAME}_emitc
+    )
+    target_compile_definitions(${_NAME}_run
+      PRIVATE
+      EMITC_IMPLEMENTATION=\"${_C_FILE_NAME}\"
+    )
+  else()
+    target_link_libraries(${_NAME}_run PRIVATE ${_NAME}_module_c)
+  endif()
 
-    string(REPLACE ";" ", " _INPUT_ENTRY_STR "${_INPUT_ENTRY_LIST}")
-    set(_INPUT_SHAPE_STR "${_INPUT_SHAPE_STR}\{${_INPUT_ENTRY_STR}\}, ")
+  add_dependencies(iree-test-deps "${_NAME}_run")
 
-    list(LENGTH _INPUT_ENTRY_LIST _INPUT_DIM)
-    list(APPEND _INPUT_DIM_LIST ${_INPUT_DIM})
-    if(_INPUT_DIM GREATER _INPUT_DIM_MAX)
-      set(_INPUT_DIM_MAX ${_INPUT_DIM})
-    endif()
-    set(_INPUT_SIZE 1)
-    foreach(_INPUT_DIM_VAL ${_INPUT_ENTRY_LIST})
-      math(EXPR _INPUT_SIZE "${_INPUT_SIZE} * ${_INPUT_DIM_VAL}")
-    endforeach()
-    list(APPEND _INPUT_SIZE_LIST ${_INPUT_SIZE})
-  endforeach(_INPUT_ENTRY)
-  string(REPLACE ";" ", " _INPUT_DIM_STR "${_INPUT_DIM_LIST}")
-  string(REPLACE ";" ", " _INPUT_SIZE_STR "${_INPUT_SIZE_LIST}")
+  iree_native_test(
+    NAME
+      ${_RULE_NAME}
+    SRC
+      ::${_RULE_NAME}_run
+    DRIVER
+      local-sync
+    LABELS
+      ${_RULE_LABELS}
+      ${_RULE_TARGET_CPU_FEATURES}
+  )
+endfunction()
 
-  # Process input and HAL data format based on the last input entry.
-  set(IREE_INPUT_TYPE)
-  set(IREE_HAL_TYPE)
-  if("${_INPUT_FORMAT}" STREQUAL "f32")
-    set(IREE_INPUT_TYPE "float")
-    set(IREE_HAL_TYPE "IREE_HAL_ELEMENT_TYPE_FLOAT_32")
-  elseif("${_INPUT_FORMAT}" STREQUAL "i32")
-    set(IREE_INPUT_TYPE "int32_t")
-    set(IREE_HAL_TYPE "IREE_HAL_ELEMENT_TYPE_SINT_32")
-  elseif("${_INPUT_FORMAT}" STREQUAL "i16")
-    set(IREE_INPUT_TYPE "int16_t")
-    set(IREE_HAL_TYPE "IREE_HAL_ELEMENT_TYPE_SINT_16")
-  elseif("${_INPUT_FORMAT}" STREQUAL "i8")
-    set(IREE_INPUT_TYPE "int8_t")
-    set(IREE_HAL_TYPE "IREE_HAL_ELEMENT_TYPE_SINT_8")
-  elseif("${_INPUT_FORMAT}" STREQUAL "ui8")
-    set(IREE_INPUT_TYPE "uint8_t")
-    set(IREE_HAL_TYPE "IREE_HAL_ELEMENT_TYPE_UINT_8")
-  else()
-    message(SEND_ERROR "Unsupported format ${_INPUT_FORMAT}")
+# iree_vmvx_linker_test
+# Creates a test test for the statically linked libraries generated by the
+# vmvx compiler target backend and executed using the local-sync runtime
+# HAL driver.
```

Comment:
This is getting too fuzzy with terminology IMO. VMVX is not statically linked. Either `build_tools/cmake/static_linker_test.c.in` and the build rule should be renamed to something else, or this VMVX coverage should reuse the other existing test setups (`iree-run-module`, `iree_check_test`).

> The goal of this feature is to potentially extend to test inlineHAL and such.

Maybe let's start there, rather than overload the terms "static linker"?